app/controllers/form_controller.rb

```diff
@@ -208,6 +217,24 @@ def successful_redirect_path
     send(redirect_path, @log)
   end
 
+  def redirect_path_to_question(page_to_show, unanswered_pages)
+    remaining_pages = unanswered_pages.excluding(page_to_show)
+    remaining_page_ids = remaining_pages.any? ? remaining_pages.map(&:id).join(",") : nil
+    send("#{@log.class.name.underscore}_#{page_to_show.id}_path", @log, { referrer: "check_answers", unanswered_pages: remaining_page_ids })
+  end
+
+  def pages_requiring_update(previously_visible_empty_page_ids)
```

Comment:
This is really the meat of the change - working out which pages require updates as a result of the submitted change and serving them up instead of going back to the CYA page. I've tried to keep it from intruding too much on the main flow, but it's a little tricky!